aws/client.go

```diff
@@ -0,0 +1,89 @@
+package aws
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/aws/transport/http"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"io"
+	"net/url"
+	"time"
+)
+
+const (
+	SignedUrlValidDuration = 30 * time.Minute // the maximum duration for S3 is 7 days
+)
+
+type Client struct {
+	s3 *s3.Client
+}
+
+func NewClient() (*Client, error) {
+	cfg, err := config.LoadDefaultConfig(context.TODO())
+	if err != nil {
+		return nil, err
+	}
+
+	return &Client{
+		s3: s3.NewFromConfig(cfg),
+	}, nil
+}
+
```

Comment:
Can you tell a bit about the `3` functions here. ? 
upload , download , presigned